### ISSUE TITLE
Tags

### ISSUE THREAD

**Issue by rimma068** [Kunzisoft/KeePassDX#633]
Please add tags support.  Maybe like an item in a three dot menu. You click on it and you see a list of all the used tags. Tags are also useful when searchable. And please put them in between Link and Notes sections in an entry view page. Thank you.

**Comment by J-Jamet** [comment]
Todo : 
- [ ] Show Tags in entries (I suggest a horizontal scrollable line with tags items below the header)
- [ ] Show Tags in groups (Show not modifiable group tags in entry)
- [ ] Add and remove tags (I suggest to make an internal pool to select Tags already created in the database)

**Comment by lohgun** [comment]
> Todo :
> 
>     * [ ]  Show Tags in entries (I suggest a horizontal scrollable line with tags items below the header)

Why not an expandable row?
 
>     * [ ]  Add and remove tags (I suggest to make an internal pool to select Tags already created in the database)

I would really like this!

Another thing could be to add an option to turn the _tree list_ into a _tag list_ where any tag is like a group plus a special "untagged" block grouping untagged entries (this will also help to keep tags organized).

**Comment by J-Jamet** [comment]
Merge in develop commit 7b5fc600f57643405cf9ef64cb5e2be9ddb56090

**Comment by aisbergde** [comment]
currently tags are not shown and it is not possible to search for them. It looks like they are completely ignored. Is it so hard to just show them and then to search content also in tags?

**Comment by J-Jamet** [comment]
Tags should be visible and editable, are you sure you are on beta 3.2.0?
The search will be done in a second time during the refactoring.

**Comment by aisbergde** [comment]
I don't use the beta, but version 3.1.0
I will try to change my settings to be able to get the beta via f-droid or Google Playstore: The latest available version is stil 3.1 0

**Comment by J-Jamet** [comment]
The beta is not deployed on F-Droid, only on Google Drive.

> The latest available version is stil 3.1 0

Yes I know, that's what we call keeping a stable version. The principle of a beta is to test elements before the production release.

> It looks like they are completely ignored. Is it so hard to just show them and then to search content also in tags?

Why ask this question if you don't even bother to see the progress of the elements, whose history is in this issue. So if you think it's not complicated for you, maybe you should make a pull request. This is an open source project and I prioritize things according to the time I have, having high expectations requires investment.

**Comment by aisbergde** [comment]
Sorry, I didn't see that it was related to a beta version which can't not yet consumed via f-droid. In the projects readme the recommendation is to use f-droid. That's why I switched to f-droid.

And it looks like it is hard to mix to get the same apps from Google Playstore and from f-droid. When Google Playstore is used as a provider, then the f-droid app will not show newer f-droid versions to download. The only way is to uninstall Google Playstore versions and then to install f-droid versions. And if I forget to save and restore settings, I need to setup again. So, I decided to use f-droid in General as main source for apps, if this is possible and not to switch.

In other words: I will wait until this version is available via f-droid. Sorry for the confusion with my comment. I just don't know, how to evaluate a beta, using the "normal" workflow.

I know that it is open source. And I can't take part in the development of this app because I am specialized in other programming languages. And therefore, I have my own open source project, which suits my knowledge.

**Comment by J-Jamet** [comment]
> The only way is to uninstall Google Playstore versions and then to install f-droid versions.

It is possible to install the F-Droid and Play Store versions in parallel because they do not have the same applicationId.
I don't know how you did it, but it's very easy, just install both.I think you just have to try.

> And if I forget to save and restore settings, I need to setup again.

The import/export system of the parameters is made on purpose and works very well.
https://github.com/Kunzisoft/KeePassDX/wiki/Import-and-Export#app-properties

> In other words: I will wait until this version is available via f-droid. Sorry for the confusion with my comment. I just don't know, how to evaluate a beta, using the "normal" workflow.

No problem, I advise people who do not want to subscribe to the beta program to stay on a stable version. But in this case, you have to be understanding about the realization of the features, the development and validation time.

> And therefore, I have my own open source project, which suits my knowledge.

Very cool, so you must understand better than anyone. ;)

**Comment by J-Jamet** [comment]
I'm closing this thread, the tags and favorites search features have their own issue.
